### PR TITLE
feat(typescript-angular): add flag for using string enums

### LINKED ISSUE #3451: [REQ][typescript-*] String Enums
### Is your feature request related to a problem? Please describe.

Since TS version 2.4 string enums are supported, which are the preferred way of defining enums that can have string values (over the legacy object-type-dualism). https://www.typescriptlang.org/docs/handbook/enums.html#string-enums

## Describe the solution you'd like

I implemented it already on a branch for internal use at my company, which is using an additional property called "stringEnums" for now. No breaking change since it needs to be enabled, off by default

## Describe alternatives you've considered

Other solutions that would be possible are:
- Default to string enums, which would be incompatible with TS<2.4 and be a breaking change
- Add a proprietary flag to the openapi spec. I think this is not really desirable because most problems that are solved this way are rather specific to the operation, whereas this is more of a global flag IMO.

## Comparison of outputs

before:

```ts
export interface User {
    title: User.TitleEnum
    name: string
}
export namespace User {
	export type TitleEnum = 'Mr' | 'Mrs' | 'Ms'
	export const TitleEnum = {
	    Mr: 'Mr' as Title,
	    Mrs: 'Mrs' as Title,
	    Ms: 'Ms' as Title,
	}
}
```

with string enums enabled:

```ts
export interface User {
    title: UserTitle
    name: string
}
export enum UserTitle {
    Mr = 'Mr',
    Mrs = 'Mrs',
    Ms = 'Ms',
}
```

### PR DESCRIPTION
**Pull request by eseliger**
### PR checklist

- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Ran the shell script under `./bin/` to update Petstore sample so that CIs can verify the change. (For instance, only need to run `./bin/{LANG}-petstore.sh`, `./bin/openapi3/{LANG}-petstore.sh` if updating the {LANG} (e.g. php, ruby, python, etc) code generator or {LANG} client's mustache templates). Windows batch files can be found in `.\bin\windows\`. If contributing template-only or documentation-only changes which will change sample output, be sure to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) first.
- [x] Filed the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master`, `4.1.x`, `5.0.x`. Default: `master`.
- [x] Copied the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) to review the pull request if your PR is targeting a particular programming language.

### Description of the PR

Closes #3451 

As discussed in the issue, adds a property to opt-in for the usage of string enums. Might be defaulted, or completely removed once we are satisfied that TS2.3 compatibility is not a requirement anymore.

@TiFu (2017/07) @taxpon (2017/07) @sebastianhaas (2017/07) @kenisteward (2017/07) @Vrolijkx (2017/09) @macjohnny (2018/01) @nicokoenig (2018/09) @topce (2018/10) @akehir (2019/07)

### CODE REVIEW

**Comment by eseliger** [comment]
is it necessary to update the `./bin/openapi3/*` files as well? Updating them yields more changes than the ones imposed by me ...

File: bin/typescript-angular-petstore-string-enums.sh [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+executable="./modules/openapi-generator-cli/target/openapi-generator-cli.jar"
+
+if [ ! -f "$executable" ]
+then
+  mvn -B clean package
+fi
+
+# if you've executed sbt assembly previously it will use that instead.
+export JAVA_OPTS="${JAVA_OPTS} -Xmx1024M -DloggerPath=conf/log4j.properties"
+ags="generate -i modules/openapi-generator/src/test/resources/2_0/petstore.yaml -g typescript-angular -o samples/client/petstore/typescript-angular-v2/string-enums --additional-properties stringEnums=true $@"
```
Comment (macjohnny) [inline bin/typescript-angular-petstore-string-enums.sh]: could you please generate this as part of the `typescript-angular-v7` samples?

**Comment by eseliger** [comment]
Okay, will do later today

**Comment by eseliger** [comment]
pardon for the interruption, I had to switch my focus on something else. I fixed it

**Comment by benwiley4000** [comment]
I'm sorry I'm such a novice here but would it be easy to port this feature to typescript-axios? The TypeScript enums are really messing up our code, which is expecting a plain string.

**Pull request merged**